### PR TITLE
Rabbitmq plugin fix

### PR DESCRIPTION
**Pull request by Yuav**
Fixes issue #575

### CODE REVIEW

File: spec/classes/collectd_init_spec.rb [hunk 1]
```diff
@@ -4,7 +4,9 @@
   on_supported_os.each do |os, facts|
     context "on #{os} " do
       let :facts do
-        facts
+        facts.merge(
+          python_dir: '/usr/local/lib/python2.7/dist-packages'
```
Comment (bastelfreak) [inline spec/classes/collectd_init_spec.rb]: can you move the fact to https://github.com/voxpupuli/puppet-collectd/blob/master/spec/default_module_facts.yml?

File: lib/facter/python_dir.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Facter.add(:python_dir) do
+  setcode do
+    Facter::Util::Resolution.exec('python -c "import site; print site.getsitepackages()[0]"')
```
Comment (bastelfreak) [inline lib/facter/python_dir.rb]: This doesn't work on my local machine:
```
bastelfreak@tmu ~ $ python2 -c "import site; print site.getsitepackages()[0]"
Error processing line 2 of /usr/lib/python2.7/site-packages/zope.app.form-4.0.2-py2.7-nspkg.pth:

  Traceback (most recent call last):
    File "/usr/lib/python2.7/site.py", line 152, in addpackage
      exec line
    File "<string>", line 1, in <module>
  KeyError: 'zope'

Remainder of file ignored
/usr/lib/python2.7/site-packages
bastelfreak@tmu ~ $ python -c "import site; print site.getsitepackages()[0]"
  File "<string>", line 1
    import site; print site.getsitepackages()[0]
                          ^
SyntaxError: invalid syntax
bastelfreak@tmu ~ $ python3 -c "import site; print site.getsitepackages()[0]"
  File "<string>", line 1
    import site; print site.getsitepackages()[0]
                          ^
SyntaxError: invalid syntax
bastelfreak@tmu ~ $
```

File: manifests/params.pp [hunk 3]
```diff
@@ -1,128 +1,131 @@
...
+      $default_python_dir = '/usr/share/collectd/python'
     }
 
     default: {
       fail("${::osfamily} is not supported.")
     }
   }
+
+  # Override with custom fact value (present only if python is installed)
+  $python_dir            = pick($::python_dir, $default_python_dir)
```
Comment (bastelfreak) [inline manifests/params.pp]: so this changes the default python directory for at least debian from a collectd specific directory to the distrowide python directory?

Comment (Yuav) [inline manifests/params.pp]: This changes the default python directory to where pip installs the package. Plugin is currently broken, as collectd is configured to look for the package where it doesn't exist (as it's installed using pip)

**Pull request merged**